tools/gulp/util/task_helpers.ts

```diff
@@ -50,33 +50,33 @@ export interface ExecTaskOptions {
   errMessage?: string;
   // Environment variables being passed to the child process.
   env?: any;
+  // Whether the task should fail if the process writes to STDERR.
+  failOnStderr?: boolean;
 }
 
 /** Create a task that executes a binary as if from the command line. */
 export function execTask(binPath: string, args: string[], options: ExecTaskOptions = {}) {
   return (done: (err?: string) => void) => {
     const env = Object.assign({}, process.env, options.env);
     const childProcess = child_process.spawn(binPath, args, {env});
+    const onClose = (code: number) => {
+      code != 0 ? done(options.errMessage || `Process failed with code ${code}`) : done();
+    };
 
     if (!options.silentStdout && !options.silent) {
       childProcess.stdout.on('data', (data: string) => process.stdout.write(data));
     }
 
-    if (!options.silent) {
-      childProcess.stderr.on('data', (data: string) => process.stderr.write(data));
-    }
-
-    childProcess.on('close', (code: number) => {
-      if (code != 0) {
-        if (options.errMessage === undefined) {
-          done('Process failed with code ' + code);
-        } else {
-          done(options.errMessage);
-        }
-      } else {
-        done();
+    childProcess.stderr.on('data', (data: string) => {
+      if (options.failOnStderr) {
+        done(data);
```

Comment:
This means that only the first `stderr` message will be printed, but there can be more `stderr` messages.

It would make more sense to exit in the `onClose` hook if there was a `stderr` output.